bayes_opt/bayesian_optimization.py

```diff
@@ -177,15 +195,17 @@ def suggest(self, utility_function):
         with warnings.catch_warnings():
             warnings.simplefilter("ignore")
             self._gp.fit(self._space.params, self._space.target)
+            if self.constraint is not None:
```

Comment:
Under the hood this also fits sklearn GPRs so I figured if we don't want to see warnings for the optimization function fitting we probably also don't want to see any for the constraint function. That being said, it would of course be simple enough to move the `fit` call out and optionally place it within its own catch_warnings context.